src/anomalib/models/__init__.py

```diff
@@ -263,7 +271,19 @@ def get_model(model: DictConfig | str | dict | Namespace, *args, **kwdargs) -> A
             model = OmegaConf.create(model)
         try:
             if len(model.class_path.split(".")) > 1:
-                module = import_module(".".join(model.class_path.split(".")[:-1]))
+                # Security check: Only allow imports from whitelisted modules
+                module_path = ".".join(model.class_path.split(".")[:-1])
+                if module_path not in ALLOWED_MODULES:
+                    logger.error(
+                        f"Module import from '{module_path}' is not allowed. "
+                        f"Only imports from {ALLOWED_MODULES} are permitted.",
+                    )
+                    msg = f"Module import from '{module_path}' is not allowed."
+                    raise UnknownModelError(msg)
+
+                # Use a whitelist approach to prevent arbitrary code execution
+                # semgrep: python.lang.security.audit.non-literal-import.non-literal-import
```

Comment:
Please use `# nosemgrep`
```suggestion
                # nosemgrep: python.lang.security.audit.non-literal-import.non-literal-import
```